### PR TITLE
refactor(detected_object_validation): apply util functions defined in the other packages

### PR DESCRIPTION
**Pull request by ktro2828**
## Description

Refactor some parts of package.
- [x] add sturct for parameter named `FilterTargetLabel` in utils.hpp https://github.com/autowarefoundation/autoware.universe/commit/23e455bfe3fd4f7146bfa66285f69c75719cf201
- [x] apply `perception_utils::isCarLikeVehicle()` function https://github.com/autowarefoundation/autoware.universe/commit/b595ae2389bf2e0003d3e7de115b6831b4a4c891
- [x] apply `tier4_autoware_utils::toPolygon2d()` function  https://github.com/autowarefoundation/autoware.universe/commit/ebf40b8f0592e9edd8c41f7d71ff53176bc718fb

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by scepter914 — COMMENTED** [review]
Thank you for well-written refactoring PR.
I just pointed out a bit, so would you confirm these?

File: perception/detected_object_validation/src/occupancy_grid_based_validator.cpp [hunk 1]
```diff
@@ -183,11 +152,9 @@ void OccupancyGridBasedValidator::showDebugImage(
   cv::Mat passed_objects_image = occ_grid.clone();
 
   // Get vehicle mask image and calculate mean within mask.
-  for (size_t i = 0; i < objects.objects.size(); ++i) {
-    const auto & object = objects.objects.at(i);
+  for (const auto & object : objects.objects) {
     const auto & label = object.classification.front().label;
-    const bool is_vehicle = Label::CAR == label || Label::TRUCK == label || Label::BUS == label ||
-                            Label::TRAILER == label;
+    const bool is_vehicle = perception_utils::isCarLikeVehicle(label);
```
[the remark this answers is not in the document]
Comment (ktro2828) [reply perception/detected_object_validation/src/occupancy_grid_based_validator.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/2093/commits/c65ef53ee69ef2ff091b90a1f2a548838513ef30

File: perception/detected_object_validation/src/obstacle_pointcloud_based_validator.cpp [hunk 2]
```diff
@@ -171,85 +172,40 @@ std::optional<size_t> ObstaclePointCloudBasedValidator::getPointCloudNumWithinPo
...
+  Polygon2d poly2d =
+    tier4_autoware_utils::toPolygon2d(object.kinematics.pose_with_covariance.pose, object.shape);
+  if (bg::is_empty(poly2d)) return std::nullopt;
+
+  pcl::PointCloud<pcl::PointXYZ>::Ptr poly3d(new pcl::PointCloud<pcl::PointXYZ>);
 
-  for (size_t i = 0; i < polygon->size(); ++i) vertices.vertices.push_back(i);
-  vertices_array.push_back(vertices);
+  size_t idx = 0;
+  for (const auto & p : poly2d.outer()) {
```
[the remark this answers is not in the document]
Comment (ktro2828) [reply perception/detected_object_validation/src/obstacle_pointcloud_based_validator.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/2093/commits/e400e4658fe1949edb5a6169cc05ea5e10144bdc

**Comment by ktro2828** [comment]
Updated to divide validation operation in `ObjectPositionFilterNode` into a method such as `ObjectLaneletFilterNode`'s `isPolygonOnverlapLanelets()` with https://github.com/autowarefoundation/autoware.universe/pull/2093/commits/b317b8170dacc40d09d81f5ea83ee674ab11d879

**Review by scepter914 — APPROVED** [review]
LGTM

**Review by yukkysaito — APPROVED** [review]
LGM

**Comment by Shin-kyoto** [comment]
https://github.com/autowarefoundation/autoware.universe/blob/a19756094d4ef32e970cca8fbf849cd3ab7d8339/perception/detected_object_validation/CMakeLists.txt#L58-L64

ここに
`src/utils.cpp`がないせいでutils.cppがbuildされずに，実行時エラーが出ます

**Pull request merged**